tests-e2e/core/dt_dtc_dtclass_test.go

```diff
@@ -0,0 +1,421 @@
+							},
+						},
+					},
+				},
+			}
+			err = k8s.Create(env, k8sClient)
+			Expect(err).To(BeNil())
+
+			By("Step 3 - ensure the DTC has a phase .status.phase of Pending, and has the correct target provisioner annotation")
+			Eventually(dtc, "60s", "1s").Should(dtcfixture.HasStatusPhase(appstudiosharedv1.DeploymentTargetClaimPhase_Pending))
+			Eventually(dtc, "2m", "1s").Should(
+				dtcfixture.HasAnnotation(appstudiosharedv1.AnnTargetProvisioner,
+					string(dtc.Spec.DeploymentTargetClassName)),
+			)
+
+			By("Step 4 - ensure that a SpaceRequest was created")
+
+			expectedSpaceRequest := codereadytoolchainv1alpha1.SpaceRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      dtc.Name,
+					Namespace: dtc.Namespace,
+				},
+				Spec: codereadytoolchainv1alpha1.SpaceRequestSpec{},
+			}
+
+			Eventually(func() bool {
+
+				spaceRequestList := codereadytoolchainv1alpha1.SpaceRequestList{}
+				err = k8s.List(&spaceRequestList, dtc.Namespace, k8sClient)
+				if err != nil {
+					fmt.Println("Error occurred in list", err)
+					return false
+				}
+				if len(spaceRequestList.Items) > 0 {
+					fmt.Println()
+				}
+
+				for _, spaceRequest := range spaceRequestList.Items {
+
+					if strings.HasPrefix(spaceRequest.Name, dtc.Name) {
+						expectedSpaceRequest = *spaceRequest.DeepCopy()
+						return true
+					} else {
+						fmt.Println("- SpaceRequest does not match:", spaceRequest.Name)
+					}
+				}
+
+				return false
+			}, "60s", "1s").Should(BeTrue())
+
+			By("Step 5 - Create a fake space request credentials secret, and update the SpaceRequest to point to that Secret, with status of Ready")
+
+			secret := corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "space-request-secret",
+					Namespace: dtc.Namespace,
+				},
+				StringData: map[string]string{
+					"some-data": generateFakeKubeConfig(),
+				},
+			}
+			err = k8s.Create(&secret, k8sClient)
+			Expect(err).To(BeNil())
+
+			err = spacerequest.UpdateStatusWithFunction(&expectedSpaceRequest, func(spaceRequestParam *codereadytoolchainv1alpha1.SpaceRequestStatus) {
+
+				*spaceRequestParam = codereadytoolchainv1alpha1.SpaceRequestStatus{
+					TargetClusterURL: "https://fake-client-api",
+					NamespaceAccess: []codereadytoolchainv1alpha1.NamespaceAccess{
+						{
+							Name:      "new-namespace",
+							SecretRef: secret.Name,
+						},
+					},
+					Conditions: []codereadytoolchainv1alpha1.Condition{
+						{
+							Type:   codereadytoolchainv1alpha1.ConditionReady,
+							Status: corev1.ConditionTrue,
+						},
+					},
+				}
+			})
+			Expect(err).To(BeNil())
+
+			By("Step 6 - Ensure there exists a DeploymentTarget based on the DTC")
+
+			var matchingDT appstudiosharedv1.DeploymentTarget
+
+			Eventually(func() bool {
+
+				deploymentTargetList := appstudiosharedv1.DeploymentTargetList{}
+				err = k8s.List(&deploymentTargetList, dtc.Namespace, k8sClient)
+				Expect(err).To(BeNil())
+
+				fmt.Println()
```

Comment:
I assume the println was used for debugging here. Should we remove it?

Reply:
The Println is just used to add a blank link between outputs of - DeploymentTarget name does not match: (this helps break up each group of iterations through the loop)